### PR TITLE
Fix clustered barcharts

### PR DESCRIPTION
**Pull request by jacoblurye**
Resolved bar-sizing issue in clustered bar charts.

Before:
screen shot 2018-08-02 at 2 37 44 pm

After:
screen shot 2018-08-02 at 2 38 33 pm

### CODE REVIEW

File: shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala [hunk 1]
```diff
@@ -90,16 +90,23 @@ object BarChart {
         val numGroups = data.map(_.cluster).distinct.size
         val barsPerGroup = if (numGroups > 1) data.groupBy(_.cluster).map(_._2.size).max else 1
 
+        val fullPlotWidth = xtransformer(plot.xbounds.min + 1.0 * numGroups)
```
Comment (triton11) [inline shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala]: can this be just `xtransformer(plot.xbounds.min + numGroups)` ?

File: shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala [hunk 2]
```diff
@@ -90,16 +90,23 @@ object BarChart {
...
         sorted.zipWithIndex
           .foldLeft(initial) {
             case ((lastCluster, d), (bar, barIndex)) =>
-              // X offset and bar width.
-              val xscale = 1.0 / barsPerGroup
-              val barx = barIndex * xscale
-              val x = xtransformer(plot.xbounds.min + barx)
-              val barWidth = xtransformer(plot.xbounds.min + barx + xscale) - x
+              // X offset
+              val clusterIndex = if (numGroups > 1) bar.cluster else barIndex
```
Comment (triton11) [inline shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala]: Idk if this is a reasonable use case, but when I use bar.cluster with just one cluster it ends up defaulting to a regular bar chart, which was a little unexpected.

File: shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala (hunk 1 above)
Comment (triton11) [inline shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala]: can this be just `xtransformer(plot.xbounds.min + numGroups)` ?

File: shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala (hunk 2 above)
Comment (jacoblurye) [reply shared/src/main/scala/com/cibo/evilplot/plot/BarChart.scala]: Agreed that that's odd behavior. I'll update so all the bars cluster above the first x label.

**Pull request merged**